File: MyFinance.Frontend/src/types/FonteTaxaJuros.ts
/**
 * De onde vem a taxa de juros anual simulada.
 * Corresponde ao enum FonteTaxaJuros.cs no backend (serializado como número).
 */
export const FonteTaxaJuros = {
    Manual: 1,
    Selic: 2,
    PercentualCdi: 3,
} as const;

export type FonteTaxaJuros = (typeof FonteTaxaJuros)[keyof typeof FonteTaxaJuros];

File: MyFinance.Domain/Enums/FonteTaxaJuros.cs
namespace MyFinance.Domain.Enums
{
    /// <summary>
    /// De onde vem a taxa de juros anual usada na projeção.
    /// </summary>
    public enum FonteTaxaJuros
    {
        /// <summary>Taxa informada manualmente pelo usuário.</summary>
        Manual = 1,

        /// <summary>Taxa Selic real vigente, buscada via Banco Central.</summary>
        Selic = 2,

        /// <summary>Percentual do CDI vigente (ex.: 110% do CDI), buscado via Banco Central.</summary>
        PercentualCdi = 3
    }
}
